src/App.jsx

```diff
@@ -20,19 +23,43 @@ export default function App() {
           console.error('error occured: ', err.message);
         });
     })();
+    if (sessionStorage.getItem('temp')) setTemperature(sessionStorage.getItem('temp'));
   }, []);
 
+  useEffect(() => {
+    const cashedTheme = sessionStorage.getItem('theme');
+    const themeElement = document.getElementById('app');
+    if (!theme) {
+      if (!!cashedTheme) {
+        themeElement.dataset.theme = cashedTheme;
+        setTheme(cashedTheme);
+      } else {
+        themeElement.dataset.theme = 'light';
+      }
+    } else {
+      sessionStorage.setItem('theme', theme);
+      document.getElementById('app').dataset.theme = theme;
```

Comment:
1. why not `themeElement.dataset.theme = theme;` instead of `document.getElementById('app').dataset.theme = theme;`?
2. as far as i see, in every condition you assign new theme for `themeElement`, so line with assignment can be moved out of condition blocks and new theme value can be defined like this:
`const newTheme = theme || cashedTheme || "light";`